sklearn/decomposition/_sparse_pca.py

```diff
@@ -307,6 +316,25 @@ class MiniBatchSparsePCA(SparsePCA):
         across multiple function calls.
         See :term:`Glossary <random_state>`.
 
+    tol : float, default=1e-3
+        Control early stopping based on the norm of the differences in the
+        dictionary between 2 steps. Used only if `max_iter` is not None.
+
+        To disable early stopping based on changes in the dictionary, set
+        `tol` to 0.0.
+
+        .. versionadded:: 1.1
+
+    max_no_improvement : int, default=10
+        Control early stopping based on the consecutive number of mini batches
+        that does not yield an improvement on the smoothed cost function. Used only if
+        `max_iter` is not None.
+
+        To disable convergence detection based on cost function, set
+        `max_no_improvement` to `None`.
+
+        .. versionadded:: 1.1
```

Comment:
In retrospect it's a bit weird to introduce a new parameter in a bugfix release.

Maybe we should instead move this PR to target 1.2 and update the previous deprecation message for the non-minibatch sparse PCA to 1.4 instead of 1.3 so that both the minibatch and non-minibatch variant are actually removed in 1.4 at the same time to ease maintenance.